internal/pkg/gateway/api.go

```diff
@@ -427,11 +430,101 @@ func (gs *Server) Submit(ctx context.Context, request *gp.SubmitRequest) (*gp.Su
 		return nil, status.Errorf(codes.Unavailable, "no orderer nodes available")
 	}
 
+	logger := logger.With("txID", request.TransactionId)
+	config := gs.getChannelConfig(request.ChannelId)
+	if config.ChannelConfig().Capabilities().ConsensusTypeBFT() {
+		return gs.submitBFT(ctx, orderers, txn, config, logger)
+	} else {
+		return gs.submitNonBFT(ctx, orderers, txn, logger)
+	}
+}
+
+func (gs *Server) submitBFT(ctx context.Context, orderers []*orderer, txn *common.Envelope, config channelconfig.Resources, logger *flogging.FabricLogger) (*gp.SubmitResponse, error) {
+	// For BFT, we send transaction to ALL orderers
+	waitCh := make(chan *gp.ErrorDetail, len(orderers))
+	go gs.broadcastToAll(orderers, txn, waitCh, logger)
+
+	n := len(orderers)
```

Comment:
if you had a config block committed during this method that changes `n`, then you will send to `n` and then analyze for `n' != n`. Let's cache `n` and the orderers and use both of them for both sending and analyzing.